### PR TITLE
Meta: added tests

### PR DESCRIPTION
**Pull request by Abb1x**
Added unit testing build support.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -20,10 +20,18 @@ LDFLAGS	+=  -lreadline						\
...
+test: CFLAGS	+=  -Iinc --coverage
+test: LDFLAGS	+= -lcmocka --coverage
+test: CC	= gcc
+
 TARGET   =  monkelisp
 SRCS     =  $(wildcard src/*.c)
 OBJS     =  $(patsubst %.c, $(BUILD_DIRECTORY)/%.c.o, $(SRCS))
 
+TEST_SRCS	= test.c
+TEST_OBJS     =  $(addprefix tests/, $(TEST_SRCS:.c=.o))
```
[the remark this answers is not in the document]
Comment (Keyboard-Slayer) [reply Makefile:32]: Can you also put the `.o`s in the build directory ?

File: Makefile [hunk 2]
```diff
@@ -35,11 +43,20 @@ build-mulib:
...
-	rm monkelisp
+	@-rm -r build
+	@-rm $(TEST_OBJS:.o=.gcno)
+	@-rm $(TEST_OBJS:.o=.d)
+	@-rm $(TEST_OBJS)
+
+	@-rm monkelisp
 	$(MAKE) -C deps/mulib clean
 
 .PHONY: all clean
```
Comment (Keyboard-Slayer) [inline Makefile:62]: You forgot to put the `test` recipe on the `.PHONY`

File: Makefile [hunk 3]
```diff
@@ -20,10 +20,18 @@ LDFLAGS	+=  -lreadline						\
 			-fsanitize=undefined			\
 			-fsanitize=address
 
+test: CFLAGS	+=  -Iinc --coverage
+test: LDFLAGS	+= -lcmocka --coverage
+test: CC	= gcc
```
Comment (Keyboard-Slayer) [inline Makefile:25]: we will also use `TCC` for the tests

**Review by Keyboard-Slayer — CHANGES_REQUESTED** [review]

**Comment by Abb1x** [comment]
Okay, I fixed everything!

**Comment by Keyboard-Slayer** [comment]
LGTM :ok_hand:

**Pull request merged**